boot/kernel_os.go

```diff
@@ -115,11 +105,13 @@ func SetNextBoot(s *snap.Info) error {
 	})
 }
 
-// ChangeRequiresReboot returns whether a reboot is required to switch
-// to the given OS, base or kernel snap.
-func ChangeRequiresReboot(s *snap.Info) bool {
-	if s.GetType() != snap.TypeKernel && s.GetType() != snap.TypeOS && s.GetType() != snap.TypeBase {
-		return false
+func (bs *coreBootParticipant) ChangeRequiresReboot() bool {
+	// XXX: at some point ChangeRequiresReboot might be in its own
```

Comment:
this a potential TODO not an XXX, s/XXX/TODO/ and put a question mark at the end?